luaui/Widgets/Shaders/map_startpolygon_gl4.frag.glsl

```diff
@@ -0,0 +1,445 @@
+	
+	#ifdef SCAV_ALLYTEAM_ID
+		int inScavBox = 0;
+	#endif
+	#ifdef RAPTOR_ALLYTEAM_ID
+		int inRaptorBox = 0;
+	#endif
+	bool isPassable = false;
+	vec3 mycolor = vec3(0,0,0);
+	#if 0
+	for (int i = 0; i < NUM_BOXES; i++) {
+		float dist = distanceToBox(mapWorldPos.xz, startBoxes[i]);
+		if (closestbox > dist){
+			closestbox = dist;
+			mycolor = teamColor[i].rgb;
+		}
+		furthestbox = max(furthestbox, dist);
+	}
+	#else
+		int startpoint = 0;
+		int teamID = int(polyVerts[startpoint].x);
+		int endpoint = 2;
+		// fair warning: there is probably a bug here that causes an infinite loop if the last box is the same team as the first box
+		// also, its not very efficient
+		// Whoever reads this code, I'm sorry :'(
+		for (int i = 0; i < NUM_POLYGONS; i = i + 1){
+			while (int(polyVerts[endpoint].x) == teamID){
+				endpoint = endpoint + 1;
+				if (endpoint == NUM_POINTS){
+					break;
+				}
+			}
+
+			float signedDistance = sdPolygon2(mapWorldPos.xz, startpoint, endpoint - startpoint);
+
+			closestbox = min(closestbox, signedDistance);
+
+			// Check if this is _our_ box
+			if (signedDistance < 0){
+				anyBoxEdgeDistance = min(anyBoxEdgeDistance, -signedDistance);
+				if (teamID == myAllyTeamID + 0){
+					mycolor.g = 0.7;
+					inAllyBox = 1;
+				}else{
+					numEnemyBoxes = numEnemyBoxes + 1;
+					mycolor.r = 1.0;
+				}
+				
+				#ifdef SCAV_ALLYTEAM_ID
+					if (teamID == SCAV_ALLYTEAM_ID){
+						inScavBox = 1;
+					}
+				#endif
+				#ifdef RAPTOR_ALLYTEAM_ID
+					if (teamID == RAPTOR_ALLYTEAM_ID){
+						inRaptorBox = 1;
+					}
+				#endif
+			}else{
+				smoothDistance = smin(smoothDistance, signedDistance, 50.0);
+
+			}
+			// Advance pointer
+			startpoint = endpoint;
+			teamID = int(polyVerts[startpoint].x);
+		}
+	#endif
+
+	// Define the colors for the individual cases: 
+
+	if (inAllyBox == 1){ // my box
+		if (numEnemyBoxes > 0){ // has enemy boxes
+			mycolor = vec3(0.5, 0.7, 0.0);
+		}else{ // solo my box
+			mycolor = vec3(0.0, 0.7, 0.0);
+		}
+	}else{ // not my box
+		if (numEnemyBoxes <= 1){ // solo enemy box
+			mycolor = vec3(1.0, 0.0, 0.0);
+			#ifdef RAPTOR_ALLYTEAM_ID
+				if (inRaptorBox == 1){
+					mycolor = vec3(1.0, 0.45, 0.0);
+				}
+			#endif
+				
+			#ifdef SCAV_ALLYTEAM_ID
+				if (inScavBox == 1){
+					mycolor = vec3(0.6, 0.0, 1.0);
+				}
+			#endif
+
+		}else{ // shared enemy box
+			mycolor = vec3(1.0, 0.2, 0.0);
+			
+			#ifdef RAPTOR_ALLYTEAM_ID
+				if (inRaptorBox == 1){
+					mycolor = vec3(1.0, 0.4, 0.0);
+				}
+			#endif
+			
+			#ifdef SCAV_ALLYTEAM_ID
+				if (inScavBox == 1){
+					mycolor = vec3(1.0, 0.3, 1.0);
+				}
+			#endif
+		}
+	}
+
+
+	// Note that now we have the distance to the closest box in closestbox
+	// and the distance to the most distant box in furthestbox
+
+	// First we color based on their distance
+	fragColor.rgba = vec4(mycolor * sin(closestbox*3 / (40/3.14)), 0.5);
+	//fragColor.rgba = vec4(mycolor*0.5, 0.5);
+	if (timeInfo.x < 150) {
```

Comment:
Consider using seconds (including below as appropriate)
```suggestion
	if (timeInfo.z < 5) {
```

Reply:
It was leftover trash anyway